sda-commons-client-jersey/src/main/java/org/sdase/commons/client/jersey/oidc/OidcClient.java

```diff
@@ -0,0 +1,84 @@
+package org.sdase.commons.client.jersey.oidc;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import java.util.Set;
+import javax.validation.*;
+import org.sdase.commons.client.jersey.ClientFactory;
+import org.sdase.commons.client.jersey.oidc.cache.AfterCreateExpiry;
+import org.sdase.commons.client.jersey.oidc.model.OidcResult;
+import org.sdase.commons.client.jersey.oidc.model.OidcState;
+import org.sdase.commons.client.jersey.oidc.rest.IssuerClient;
+import org.sdase.commons.client.jersey.oidc.rest.model.TokenResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Service that can retrieve a new access token. */
+public class OidcClient {
+  private static final Logger LOGGER = LoggerFactory.getLogger(OidcClient.class);
+  private static final String CACHE_KEY = "oidcResultKey";
+
+  private final OidcConfiguration config;
+  private final Cache<String, TokenResource> cache;
+  private final IssuerClient issuerClient;
+
+  public OidcClient(ClientFactory clientFactory, OidcConfiguration config) {
+    this.config = config;
+
+    try (ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory()) {
+      Validator validator = validatorFactory.getValidator();
+      Set<ConstraintViolation<OidcConfiguration>> violations = validator.validate(config);
+
+      if (!violations.isEmpty()) {
+        throw new ConstraintViolationException(violations);
+      }
+    }
```

Comment:
Is this really necessary? I think Dropwizard will already do that for us. If the config is invalid the app will not start up. Of course, users of the OIDC client need to add `@Valid` to the connfiguration in their configuration class.

Reply:
Right, with `@Valid ` in the configuration we don't need this manual validation, I'll remove it.